### PR TITLE
Redis on Ubuntu 14.04 LTS Guide

### PR DESCRIPTION
**Pull request by spariev**
This is mostly update for Redis on Ubuntu 12.04 Guide (https://github.com/linode/docs/blob/master/docs/databases/redis/redis-on-ubuntu-12-04-precise-pangolin.md)

### CODE REVIEW

File: docs/databases/redis/redis-on-ubuntu-14-04.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+Each option has it's own pros and cons, which are described in great detail in Redis documentation. For most possible data safety it is advised to use both persistence methods.
+
+As the point-in-time snapshot persistence is enabled by default, the only thing you will need to do is to setup AOF persistence.
+For this, add the following lines to `redis.conf` file:
+
+{: .file-excerpt }
+/etc/redis/redis.conf
+:   ~~~
+    appendonly yes
```
Comment (alexfornuto) [inline docs/databases/redis/redis-on-ubuntu-14-04.md]: per @EdwardAngert: AppendOnly is already in the /etc/redis/redis.conf file

File: docs/databases/redis/redis-on-ubuntu-14-04.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+to change overcommit memory setting immediately. To make the change permanent, add line `vm.overcommit_memory = 1` to `/etc/sysctl.conf` -
+
+{: .file-excerpt }
+/etc/sysctl.conf
+:   ~~~
+    vm.overcommit_memory = 1
+    ~~~
+
+2.  Disable Linux kernel feature transparent huge pages, as this feature will affect Redis performance in a negative way. Run
```
Comment (alexfornuto) [inline docs/databases/redis/redis-on-ubuntu-14-04.md]: per @EdwardAngert  "/sys/kernel/mm/transparent_hugepage/enabled" No such file or directory

File: docs/databases/redis/redis-on-ubuntu-14-04.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+The `slaveof` directive takes two arguments: the first is the IP address of the master node; the second is the Redis port specified in the master's configuration.
+
+Restart the slave Redis instance:
+
+	    sudo service redis-server restart
+
+After restart slave instance will attempt to synchronize its data set to the master and then propagate the changes.
+
+5.  Test that the replication works. On master Linode, run `redis-cli` and execute command `set 'a' 1`
```
Comment (alexfornuto) [inline docs/databases/redis/redis-on-ubuntu-14-04.md]: Per @EdwardAngert: Step 5: Read only. "Can't write against a read only slave"

File: docs/databases/redis/redis-on-ubuntu-14-04.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+  name: Sergey Pariev
+  link: https://twitter.com/spariev
+external_resources:
+ - '[Redis Project Home Page](http://redis.io/)'
+ - '[Redis Configuration](http://redis.io/topics/config)'
+ - '[Redis Persistence](http://redis.io/topics/persistence)'
+ - '[Redis Security](http://redis.io/security)'
+---
+
+Redis is an open source in-memory data structure store, with optional disk writes for persistence, which can be used as key-value database, cache and message broker. Redis features built-it transactions, replication, and support for a variety of data structures such as strings, hashes, lists, sets and others. Redis can be made highly available with Redis Sentinel and supports automatic partitioning with Redis Cluster. This document provides both instructions for deploying the Redis server and an overview of best practices for maintaining Redis instances.
```
Comment (alexfornuto) [inline docs/databases/redis/redis-on-ubuntu-14-04.md]: built-it -> built-in

**Comment by alexfornuto** [comment]
@EdwardAngert's tech test is complete, issues have been noted above.

File: docs/databases/redis/redis-on-ubuntu-14-04.md (hunk 3 above)
Comment (spariev) [inline docs/databases/redis/redis-on-ubuntu-14-04.md]: @EdwardAngert are you sure you executed this command on master Linode ? Slave node is indeed read-only by default, and I cant reproduce this error on master node.

File: docs/databases/redis/redis-on-ubuntu-14-04.md (hunk 2 above)
Comment (spariev) [inline docs/databases/redis/redis-on-ubuntu-14-04.md]: I'll remove this one, as `transparent hugepage` feature was apparently removed from custom linode kernels. I'm sure it did work back in January though.

File: docs/databases/redis/redis-on-ubuntu-14-04.md (hunk 1 above)
Comment (spariev) [inline docs/databases/redis/redis-on-ubuntu-14-04.md]: I'll reword this section, in lines of 
> make sure `appendonly` setting is set to `yes` and `appendfsync` is set to  `everysec`

as these settings are indeed present in default configuration file, but with other values.

File: docs/databases/redis/redis-on-ubuntu-14-04.md (hunk 3 above)
Comment (spariev) [inline docs/databases/redis/redis-on-ubuntu-14-04.md]: @alexfornuto , @EdwardAngert can you please look into it one more time ? I've checked again and on master node everything works fine.

**Comment by EdwardAngert** [comment]
Thanks Sergey,
I'll run through it again and get back to you with the results

File: docs/databases/redis/redis-on-ubuntu-14-04.md (hunk 3 above)
Comment (EdwardAngert) [inline docs/databases/redis/redis-on-ubuntu-14-04.md]: Thanks Sergey,
I'll run through it again and get back to you with the results

**Comment by spariev** [comment]
@EdwardAngert , @alexfornuto I've merged Edward's edits regarding master/slave setup.

File: docs/databases/redis/redis-on-ubuntu-14-04.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+>This guide is written for a non-root user. Commands that require elevated privileges are prefixed with `sudo`. If you’re not familiar with the `sudo` command, you can check our [Users and Groups](/docs/tools-reference/linux-users-and-groups) guide.
+
+
+## Redis Installation
+
+Redis package in Ubuntu 14.04 repository is outdated and misses several security patches, so in this case it is better to use third-party PPA.
+
+1.  Add Redis PPA repository to install the latest version:
+
+		sudo add-apt-repository ppa:chris-lea/redis-server
```
Comment (alexfornuto) [inline docs/databases/redis/redis-on-ubuntu-14-04.md]: I did some searching online, and found several references for the PPA `ppa:rwky/redis`. Can you explain the difference between this and the PPA you use in the guide?

Comment (alexfornuto) [inline docs/databases/redis/redis-on-ubuntu-14-04.md]: Answered my own question:

```
This ppa is no longer maintained. Please consider https://launchpad.net/~chris-lea/+archive/ubuntu/redis-server instead
```

File: docs/databases/redis/redis-on-ubuntu-14-04.md [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+## Redis Configuration
+
+### Configure Persistence Options
+
+Redis provides two options for disk persistence:
+
+* Point-in-time snapshots of the dataset, made at specified intervals (RDB)
+* Append only logs of all the write operations performed by the server (AOF).
+
+Each option has it's own pros and cons, which are described in great detail in Redis documentation. For most possible data safety it is advised to use both persistence methods.
```
Comment (alexfornuto) [inline docs/databases/redis/redis-on-ubuntu-14-04.md]: Can you link to said documentation?

**Comment by alexfornuto** [comment]
@spariev The guide looks good so far! Please review and merge or discuss https://github.com/spariev/docs/pull/4, after which we'll perform a final copy edit and move on to the publishing steps.

**Comment by spariev** [comment]
@alexfornuto I've merged your edits, thanks!

**Comment by alexfornuto** [comment]
@spariev I think we'll be making a couple adjustments on Monday, and then publishing! Remind me, did you prefer Linode credit or Paypal, and what email address should we use?

**Comment by spariev** [comment]
@alexfornuto great! I'd prefer PayPal, email is serhiy.pariyev@gmail.com

**Comment by EdwardAngert** [comment]
https://github.com/spariev/docs/pull/6
New pull request - check it out and let us know if you approve of the added section

**Comment by spariev** [comment]
@alexfornuto , @EdwardAngert  I've merged the changes, thanks for the fixes.

**Comment by alexfornuto** [comment]
OK, I don't know why this still shows as open, but As previously discussed this has been published. Thanks for your contribution!

**Pull request closed without merging**